tree/tree/src/TLeafF16.cxx

```diff
@@ -218,8 +220,19 @@ void TLeafF16::Streamer(TBuffer &R__b)
    if (R__b.IsReading()) {
       R__b.ReadClassBuffer(TLeafF16::Class(), this);
 
-      if (fTitle.Contains("["))
-	 fElement = new TStreamerElement(Form("%s_Element", fName.Data()), fTitle.Data(), 0, 0, "Float16_t");
+      if (fTitle.Contains("/[")) {
+         auto slash = fTitle.First("/");
+         TString bracket = slash == TString::kNPOS ? fTitle : fTitle(slash + 1, fTitle.Length() - slash - 1);
+         fElement = new TStreamerElement(Form("%s_Element", fName.Data()), bracket.Data(), 0, 0, "Float16_t");
+      } else if (R__b.Class_Version() < 2) {
```

Comment:
This version number is the in memory version of the `TBufferFile` class.  I.e. not what we are looking for.   You need to have `ReadClassBuffer` tell us the actual onfile version of the object read:
```
         R__b.ReadClassBuffer(TLeafF16::Class(), this, R__v, R__s, R__c);
```
(see `TTree::Streamer`)